### PR TITLE
Added configurationProfileAssignmentIntent with new api version 2021-03-30-preview

### PR DESCRIPTION
**Pull request by ikanni**
…03-30-preview

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [X] new API version for a new resource
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet. **=> April 2021**
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet. **=> March 2021**
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [X] No, no need to refresh for updates in this PR

### Contribution checklist:
- [X] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [X] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [X] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json
Comment (Bpoe) [inline specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json:132]: Missing 404

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/examples/listConfigurationProfileAssignmentIntentsByResourceGroup.json
Comment (Bpoe) [inline specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/examples/listConfigurationProfileAssignmentIntentsByResourceGroup.json:12]: response should be:
{
  "value": []
}

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/examples/listConfigurationProfileAssignmentIntentsBySubscription.json
Comment (Bpoe) [inline specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/examples/listConfigurationProfileAssignmentIntentsBySubscription.json:10]: Response should be:
{
   "values": []
}

**Review by Bpoe — CHANGES_REQUESTED** [review]
Small changes needed

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json
Comment (pilor) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json:132]: all errors should fall through to "default"

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json
Comment (pilor) [inline specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json:303]: This is almost identical to configurationProfileAssignments.  What is the difference and why do you need both?

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json
Comment (pilor) [inline specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json:154]: You should be able to list the configurationProfileAssignments (and configurationProfileAssignmentIntents) on a single VM as well. Right now the user has to list at the RG level then try to filter down to just the VM they care about to see eveyrthing on one VM.

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json
Comment (pilor) [inline specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json:100]: put and get but no delete?  How does someone remove an intent that they created?

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json
Comment (ikanni) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json:132]: Thank you both for reviewing this PR!

@Bpoe, Based on @pilor comment, I assume the current default is good enough. I will not change.

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/examples/listConfigurationProfileAssignmentIntentsByResourceGroup.json
Comment (ikanni) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/examples/listConfigurationProfileAssignmentIntentsByResourceGroup.json]: Thank you! Resolved!

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/examples/listConfigurationProfileAssignmentIntentsBySubscription.json
Comment (ikanni) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/examples/listConfigurationProfileAssignmentIntentsBySubscription.json]: Thank you! Resolved!

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json
Comment (ikanni) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json:303]: Thank you Chris for noticing and asking question!

1) configurationProfileAssignments has the property **"x-ms-long-running-operation": true**. It is long running operation. So it has the provisioningState as per the ARM guidelines.

2) configurationProfileAssignmentIntents has the property **"x-ms-long-running-operation": false**. It is not long running. When customers call the PUT configurationProfileAssignmentIntents, it do a PUT on configurationProfileAssignments  and return immediately.

We are integrating Automanage with CreateVM flow in portal UX. When we use configurationProfileAssignments it takes about 10+ min to reach a terminal state and deployment waits for the terminal state. As per the compute portal team expectation or standards they cannot wait for that long. 

**Once we complete the implementation of configurationProfileAssignmentIntents we will use that in CreateVM flow in portal UX. So we complete the operation in 4 sec or less.**

Comment (ikanni) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json:303]: Related post: https://stackoverflow.microsoft.com/questions/248238

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json
Comment (ikanni) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json:100]: When an configurationProfileAssignments is deleted we will delete the related configurationProfileAssignmentIntents too. That is why we did not implement the delete.

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json
Comment (ikanni) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json:154]: We will have one configurationProfileAssignments (and configurationProfileAssignmentIntents) on a single VM with the name **default**.

We will not multiple configurationProfileAssignments (and configurationProfileAssignmentIntents) on a single VM. So the list  the configurationProfileAssignments (and configurationProfileAssignmentIntents) on a single VM is not required.

Comment (ikanni) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json:154]: I hope I am clear on the above reply.

Comment (jorgecotillo) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json:154]: I don't quite follow your response. You are saying that you would not have multiple configurationProfileAssignmentIntents but your PUT operation above allows to create multiple because you have this: configurationProfileAssignmentIntentName if your intent is to always have one, then you should replace: /{configurationProfileAssignmentIntentName} with /default to denote a singleton. Also, where's your configurationProfileAssignments definition?

---
In reply to: [603716726](https://github.com/Azure/azure-rest-api-specs/pull/13556#discussion_r603716726) [](ancestors = 603716726)

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json
Comment (jorgecotillo) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json:100]: Can we get your manager's comment regarding this design choice please? You are violating our guideline, see: [Incosistent or Incomplete Crud](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review)

---
In reply to: [603708234](https://github.com/Azure/azure-rest-api-specs/pull/13556#discussion_r603708234) [](ancestors = 603708234)

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json
Comment (jorgecotillo) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json:303]: How does a user know that the provisioning is complete? Say that a user calls your API, how can they poll to get the provisioningState? 

---
In reply to: [603703789](https://github.com/Azure/azure-rest-api-specs/pull/13556#discussion_r603703789) [](ancestors = 603703789)

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json
Comment (ikanni) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json:100]: Please scan comments in https://stackoverflow.microsoft.com/questions/216549

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json
Comment (ikanni) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json:154]: Thank you for this comment! It is valid question. I will reply after discussing with our team.

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json
Comment (ikanni) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json:303]: Thank you!

@jorgecotillo, I assume you are asking the above question for configurationProfileAssignmentIntents.

The configurationProfileAssignmentIntents resource will do synchronous operation and it will not have the provisioningState property.  If there is no provisioningState then ARM will assume as succeeded and it will not poll.

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json
Comment (ikanni) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json:140]: Fixed this! Thank you!

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json
Comment (jorgecotillo) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json:93]: Got it, if this proxy resource, you are good, thanks!

---
In reply to: [605165563](https://github.com/Azure/azure-rest-api-specs/pull/13556#discussion_r605165563) [](ancestors = 605165563)

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json
Comment (jorgecotillo) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json:289]: Yeah, I think I get that part, but if I am a user, and I make an api call to this intent endpoint, how do I know if configurationProfileAssignment provisioning succeeded/failed? This intent seems like a fire and forget type of situation where returns fast and internally does a PUT on configurationProfileAssignment, I think you'll need to document this pretty well, is that your plan to avoid customer confusion?

---
In reply to: [605302690](https://github.com/Azure/azure-rest-api-specs/pull/13556#discussion_r605302690) [](ancestors = 605302690)

Comment (ikanni) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json:289]: Thank you Jorge! It is valid question.

We have two scenarios

1. **SDK/API** : We need to add clear documentation for this as you have mentioned. Thank you for pointing this out. To add documentation opened  [**Task 9651989: [Doc]ConfigurationProfileAssignmentIntents**](https://msazure.visualstudio.com/One/_workitems/edit/9651989)
2.  **Portal** : When we consume the configurationProfileAssignmentIntents PUT operation in create VM flow. The customers will navigate to VM  blade and we have Automanage (in table of contents) which already fetches the configurationProfileAssignment . So portal should be good.

**Comment by ikanni** [comment]
@ArcturusZhang, Could you please do the needful to merge this PR ?  Let me know if you need anything else from my end.

**Review by hassantsyedmsft — APPROVED** [review]

**Review by ArcturusZhang — APPROVED** [review]

**Comment by ikanni** [comment]
@ArcturusZhang Thank you for the approval! Do I need to do anything for this to be merged ?

**Comment by ArcturusZhang** [comment]
@msyyc could you please take a look at the python generation failure?

**Comment by ikanni** [comment]
@msyyc  @ArcturusZhang, Any Updates ?

**Comment by msyyc** [comment]
python pipeline failure have been fixed.

**Review by ArcturusZhang — APPROVED** [review]

**Comment by ArcturusZhang** [comment]
> Hi @ikanni, one or multiple breaking change(s) is detected in your PR. Please check out the breaking change(s), and provide business justification in the PR comment and @ PR assignee why you must have these change(s), and how external customer impact can be mitigated. Please ensure to follow [breaking change policy](https://aka.ms/AzBreakingChangesPolicy) to request breaking change review and approval before proceeding swagger PR review.
> **Action**: To initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).
> If you want to know the production traffic statistic, please see [ARM Traffic statistic](https://portal.azure-devex-tools.com/api/pr/query-traffic?pr=Azure/azure-rest-api-specs/13556&days=5).
> If you think it is false positive breaking change, please provide the reasons in the PR comment, report to Swagger Tooling Team via https://aka.ms/swaggerfeedback.

@ikanni could you please follow the above instructions to open a breaking change review request? Once we get approval for this, we could merge this PR

**Comment by ikanni** [comment]
When I started this PR  **Cross Version BreakingChange (Base on preview version)** was a warning and now it is marked as error.  I am not sure what changed as error.

@ArcturusZhang In our scenario we are going to maintain both preview version _2020-06-30-preview_ and _2021-03-30-preview_.  

1. In _2020-06-30-preview_  version we have the 3 resources 1) **Accounts**, 2) **ConfigurationProfilePreferences** 3) **ConfigurationProfileAssignments**.
2. In _2021-03-30-preview_, we have added new resource named **configurationProfileAssignmentIntents**

By considering the above scenario, **could you  please approve and merge** ?

**Comment by ArcturusZhang** [comment]
Hey @ikanni, is there any update on the brekaing change review?

**Comment by JackTn** [comment]
Hi, @ikanni. The PR has be closed for a long time and it's related branch still exist. Please tell me if you still need this branch or i will delete it in 14 days.

**Pull request closed without merging**
